### PR TITLE
[Keyboard] update xs60 to use rgb matrix 

### PR DESCRIPTION
**Pull request by Xelus22**
## Description

Update to use RGB matrix.

Move things into info.json

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* move to info.json

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
> keyboard updates and refactors (eg. to data driven) must go through develop to reduce master -> develop merge conflicts

and also any additional implementations of"VIA indicators" are currently not being accepted to the repo.

**Comment by Xelus22** [comment]
> > keyboard updates and refactors (eg. to data driven) must go through develop to reduce master -> develop merge conflicts
> 
> and also any additional implementations of"VIA indicators" are currently not being accepted to the repo.

Removed VIA indicators code

**Review by drashna — APPROVED** [review]

File: keyboards/xelus/xs60/soldered/rules.mk [hunk 1]
```diff
@@ -1,18 +1,8 @@
...
-MOUSEKEY_ENABLE = yes       # Mouse keys
-EXTRAKEY_ENABLE = yes       # Audio control and System control
-CONSOLE_ENABLE = yes        # Console for debug
-COMMAND_ENABLE = no         # Commands for debug and configuration
-NKRO_ENABLE = yes           # Enable N-Key Rollover
-BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
-AUDIO_ENABLE = no           # Audio output
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+WS2812_DRIVER = pwm
```
Comment (fauxpark) [inline keyboards/xelus/xs60/soldered/rules.mk:6]: These can be moved to info.json.

File: keyboards/xelus/xs60/hotswap/rules.mk [hunk 2]
```diff
@@ -1,18 +1,8 @@
...
-MOUSEKEY_ENABLE = yes       # Mouse keys
-EXTRAKEY_ENABLE = yes       # Audio control and System control
-CONSOLE_ENABLE = yes        # Console for debug
-COMMAND_ENABLE = no         # Commands for debug and configuration
-NKRO_ENABLE = yes           # Enable N-Key Rollover
-BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
-AUDIO_ENABLE = no           # Audio output
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+WS2812_DRIVER = pwm
```
Comment (fauxpark) [inline keyboards/xelus/xs60/hotswap/rules.mk:6]: As above.

File: keyboards/xelus/xs60/hotswap/hotswap.c [hunk 3]
```diff
@@ -18,3 +18,18 @@
 
 // tested and working
 void matrix_io_delay(void) { __asm__ volatile("nop\nnop\nnop\n"); }
+
+led_config_t g_led_config = { {
```
Comment (fauxpark) [inline keyboards/xelus/xs60/hotswap/hotswap.c:22]: As above.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Pull request closed without merging**